allennlp/commands/predict.py

```diff
@@ -171,14 +172,17 @@ def _get_instance_data(self) -> Iterator[Instance]:
 
     def run(self) -> None:
         has_reader = self._dataset_reader is not None
+        index = 0
```

Comment:
you may find this too cute, but one thing you could do to avoid having to track and increment the indices would be

```
index = itertools.count()
```

which generates an infinite sequence 0, 1, 2, ...

and then grab its value like

```
self._maybe_print_to_console_and_file(next(index), result, ...)
```

Reply:
seems fine to me, as long as it's clear what's happening